sanic_ext/extensions/templating/render.py

```diff
@@ -43,22 +43,30 @@ async def render(
     app: Optional[Sanic] = None,
     environment: Optional[Environment] = None,
     context: Optional[Dict[str, Any]] = None,
+    *,
+    template_source: str = ""
 ) -> HTTPResponse:
     if app is None:
         try:
             app = Sanic.get_app()
         except SanicException as e:
             raise SanicException(
-                "Cannot render template beause locating the Sanic application "
-                "was ambiguous. Please return  render(..., app=some_app)."
+                    "Cannot render template beause locating the Sanic application "
+                    "was ambiguous. Please return  render(..., app=some_app)."
             ) from e
 
+    if template_name and template_source:
+        raise SanicException(
+                "You must provide template_name OR template_source."
```

Comment:
Let's be explicit that they provided both.

```suggestion
                "You must provide template_name OR template_source, not both."
```